internal/querynodev2/services.go

```diff
@@ -316,6 +316,9 @@ func (node *QueryNode) WatchDmChannels(ctx context.Context, req *querypb.WatchDm
 	}
 	err = loadGrowingSegments(ctx, delegator, req)
 	if err != nil {
+		// remove legacy growing
+		node.manager.Segment.RemoveBy(segments.WithChannel(channel.GetChannelName()),
```

Comment:
Not onlyt do we need to revert growing segments, but also L0 Segments.

in fact, if any part of the watch failed, we should remove did the right the cleanup.

node manager colelction reference might leaked as well.

	defer func() {
		if err != nil {
		node.manager.Segment.RemoveBy(segments.WithChannel(channel.GetChannelName()),segments.WithType(segments.SegmentTypeGrowing)
		}
	}()

Reply:
Level zero segment will be release since it appears in segment distribution(all sealed ones do)
Handling collection ref here